internal/services/communication/email_service_domain_resource.go

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package communication
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/communication/2023-03-31/domains"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/communication/helper"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/communication/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.ResourceWithUpdate = EmailCommunicationServiceDomainResource{}
+
+type EmailCommunicationServiceDomainResource struct{}
+
+type EmailCommunicationServiceDomainResourceModel struct {
+	Name                   string            `tfschema:"name"`
+	ResourceGroupName      string            `tfschema:"resource_group_name"`
+	EMailServiceName       string            `tfschema:"email_service_name"`
+	DomainManagement       string            `tfschema:"domain_management"`
+	UserEngagementTracking bool              `tfschema:"user_engagement_tracking"`
+	Tags                   map[string]string `tfschema:"tags"`
+
+	FromSenderDomain     string                           `tfschema:"from_sender_domain"`
+	MailFromSenderDomain string                           `tfschema:"mail_from_sender_domain"`
+	VerificationRecords  []EmailDomainVerificationRecords `tfschema:"verification_records"`
+}
+
+type EmailDomainVerificationRecords struct {
+	Domain []helper.DomainVerificationRecords `tfschema:"domain"`
+	SPF    []helper.DomainVerificationRecords `tfschema:"spf"`
+	DMARC  []helper.DomainVerificationRecords `tfschema:"dmarc"`
+	DKIM   []helper.DomainVerificationRecords `tfschema:"dkim"`
+	DKIM2  []helper.DomainVerificationRecords `tfschema:"dkim2"`
+}
+
+func (EmailCommunicationServiceDomainResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"email_service_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.CommunicationServiceName,
+		},
+
+		"domain_management": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringInSlice(domains.PossibleValuesForDomainManagement(), false),
+		},
+
+		"user_engagement_tracking": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Computed: true,
+		},
```

Comment:
I search through the whole provider and I can't fine a schema combination with Default and Required. All search hits reports `Default` in combination with `Optional`. 

Mark a variables as required with default value (which make the input field not longer required feels wrong.

Reply:
Sorry for the confusion. Schema cannot have `Default` and `Required` set, my comment was to suggest one or the other. In cases where the Service "default" can change depending on external factors (sadly there are a number of cases of this) then we tend to have to make the field `Required`. However, if this always default to `false` it can be left as `Optional`. Since the implicit default value for a `bool` is `false`, the schema `Default` can be omitted as above.